hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/CryptoTransferHandler.java

```diff
@@ -195,7 +196,10 @@ private void warmNftTransfer(
         nftTransfer.ifReceiverAccountID(receiverAccountID -> {
             final Account receiver = accountStore.getAliasedAccountById(receiverAccountID);
             if (receiver != null) {
-                receiver.ifHeadTokenId(headTokenID -> tokenRelationStore.warm(receiverAccountID, headTokenID));
+                receiver.ifHeadTokenId(headTokenID -> {
```

Comment:
The `receiver`'s head `tokenRelation` will only be updated it is being auto-associated to the token type of the NFT. 

And the token type pointed to by `headTokenID` should never be used during a `CryptoTransfer`, unless by coincidence it appears in another `tokenTransferList` in the same transaction.

If these warmings are empirically helpful, LGTM...just curious why they would be. 🤔

Reply:
Seems like the test was doing auto-associations and without warming `headTokenID` it was causing performance issue.

Reply:
Well, that's exactly the case with the benchmark: it relies on auto-association as otherwise there would need to be billions of API calls to associate all tokens to their receiving accounts, which is impractical (would take many days at 10K TPS). I realize it's not reflective of real-life workloads but it may be considered the worst case scenario and thus methodologically acceptable (we know we'll get better TPS with pre-associated tokens).